url/Cargo.toml

```diff
@@ -22,17 +22,25 @@ serde_json = "1.0"
 bencher = "0.1"
 
 [dependencies]
-form_urlencoded = { version = "1.2.1", path = "../form_urlencoded" }
-idna = { version = "0.5.0", path = "../idna" }
-percent-encoding = { version = "2.3.1", path = "../percent_encoding" }
-serde = { version = "1.0", optional = true, features = ["derive"] }
+form_urlencoded = { version = "1.2.1", path = "../form_urlencoded", default-features = false, features = ["alloc"] }
+idna = { version = "0.5.0", path = "../idna", default-features = false, features = ["alloc"] }
+percent-encoding = { version = "2.3.1", path = "../percent_encoding", default-features = false, features = ["alloc"] }
+serde = {version = "1.0", optional = true, features = ["derive"]}
+no-std-net = { version = "0.6.0", default-features = false, optional = true }
 
 [features]
-default = []
+default = ["std"]
```

Comment:
I fear this might be a fairly big breaking change?

When I added `no_std` support for the other libraries in this crate, it was more trivial, since [there was only three crates that disabled default features](https://github.com/servo/rust-url/pull/722#pullrequestreview-1306599961) - but that is not the case for `url`; there are _many_ crates specifying `default-features = false`, so even though there _was_ no default features previously, this change will affect all of those crates.

<details>
<summary>Python script for finding crates using `default-features = false`.</summary>

```python
import json
import requests
import sys

API_URL = "https://crates.io/api/v1/crates"

crate_name = sys.argv[1]

page = 1
while True:
    # print(f"page {page}")
    data = json.loads(requests.get(f"{API_URL}/{crate_name}/reverse_dependencies?per_page=100&page={page}").content)
    page += 1

    if len(data['versions']) == 0:
        break

    versions = {x['id']: x for x in data['versions']}

    for dep in data['dependencies']:
        if not dep['default_features']:
            crate = versions[dep['version_id']]
            print(f"{crate['crate']} had `default-features = false`")
```
</details>

Reply:
I don't see any other way?

Reply:
Well... I see two:
- Bundling this in with other breaking changes, and releasing `url v0.3`
- Calling the feature `unstable-no-std` instead, to signify that it may break other crates in the dependency chain, and that it is a bad way of doing it which will be rectified once `v0.3` is released.